consoleme/handlers/v2/typeahead.py

```diff
@@ -97,3 +100,58 @@ async def get(self):
             self.write(json.dumps([{"title": arn} for arn in arn_array.__root__]))
         else:
             self.write(arn_array.json())
+
+
+class SelfServiceStep1ResourceTypeahead(BaseAPIV2Handler):
+    async def get(self):
+        try:
+            type_ahead: Optional[str] = (
+                self.request.arguments.get("typeahead")[0].decode("utf-8").lower()
+            )
+        except TypeError:
+            type_ahead = None
+        if not type_ahead:
+            self.write(json.dumps([]))
+            return
+        try:
```

Comment:
This block took me a bit to comprehend since it relies on some not-super-obvious behaviors in Python. It appears to be safe, but could stand to be a little more verbose for understandability.